event/type.go

```diff
@@ -0,0 +1,33 @@
+package event
+
+// Type represents the different event types emitted by k6.
+type Type string
```

Comment:
Could it be a non-string type?

Reply:
Why?

I considered making it an enum, but then we'd have to use enumer and `go generate` and that whole mess, and it really makes no difference in this case. We just need something unique and stringifiable, and plain string consts work fine IMO.

Reply:
Just because something like `uint8` can hold all variations of the types and has a smaller memory footprint, remember that the primary use case of using this is inside the go code.

I fully agree with you that following the current approach with enums sometimes looks scary (e.g. https://github.com/grafana/k6/blob/master/lib/compatibility_mode_gen.go), and just implementing `Stringer` method is enough.

But for sure, not forcing, but just sharing my opinion :relaxed:

Reply:
Fair point, but I'm not certain if memory usage is a concern. Even with 2 events emitted per iteration, events are either consumed, or dropped if there are no consumers or the event channel buffer is full. So at most the size of the channel buffer will be kept in memory, everything else will be discarded.

But sure, it's not much work to switch to an enum, and it would be safer, so I'll do that.